app/src/main/java/org/gdg/frisbee/android/event/EventActivity.java

```diff
@@ -53,16 +65,65 @@ public void onCreate(Bundle savedInstanceState) {
         mViewPager.setAdapter(mViewPagerAdapter);
         mTabLayout.setupWithViewPager(mViewPager);
 
-        mEventId = getIntent().getStringExtra(Const.EXTRA_EVENT_ID);
+        mEventId = getEventIdFrom(getIntent());
         String section = getIntent().getStringExtra(Const.EXTRA_SECTION);
         if (EventPagerAdapter.SECTION_OVERVIEW.equals(section)) {
             mViewPager.setCurrentItem(0);
         }
     }
 
+    private String getEventIdFrom(Intent intent) {
+        return intent.getStringExtra(Const.EXTRA_EVENT_ID);
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        recordStartPageView();
+    }
+
+    @Override
+    protected void onStop() {
+        recordEndPageView();
+        super.onStop();
+    }
+
     protected String getTrackedViewName() {
         return "Event/" + getResources().getStringArray(R.array.event_tabs)[getCurrentPage()] 
-                + "/" + getIntent().getStringExtra(Const.EXTRA_EVENT_ID);
+                + "/" + getEventIdFrom(getIntent());
+    }
+
+    private void recordStartPageView() {
+        App.getInstance().getGdgXHub().getEventDetail(
```

Comment:
I thought I can get away with it for this PR. Here is my suggestion for handling the communication between activity and fragment: da27744b742a92295596397c2990d60393a0c687